qark/scanner/scanner.py

```diff
@@ -28,6 +31,9 @@ def __init__(self, decompiler):
         :param Decompiler decompiler: the decompiler class that contains decompiled path information
         """
         self.decompiler = decompiler
+        if decompiler.source_code:
```

Comment:
Isn't this logic already in the decompile() function?

Reply:
Good find, I think with the other changes that this code is useless.